### PR TITLE
Add keycloak auth

### PR DESCRIPTION
**Pull request by rhuanbarreto**
For parse-community/parse-server#6376

### CODE REVIEW

File: _includes/parse-server/third-party-auth.md [hunk 1]
```diff
@@ -189,6 +190,37 @@ Google oauth supports validation of id_token's and access_token's.
...
+}
+```
+
+The authentication module will test if the authData is the same as the
+userinfo oauth call, comparing the attributes
+
+Copy the JSON config file generated on Keycloak ([how-to link](https://www.keycloak.org/docs/latest/securing_apps/index.html#_javascript_adapter))
+and paste it inside of a folder (Ex.: `auth/keycloak.json`) in your server.
+
+The options passed to Parse server:
```
Comment (TomWFox) [inline _includes/parse-server/third-party-auth.md]:
```suggestion
The options passed to Parse Server:
```

File: _includes/parse-server/third-party-auth.md [hunk 2]
```diff
@@ -189,6 +190,37 @@ Google oauth supports validation of id_token's and access_token's.
...
+    "access_token": "access token you got from keycloak JS client authentication",
+    "id": "the id retrieved from client authentication in Keycloak",
+    "roles": ["the roles retrieved from client authentication in Keycloak"],
+    "groups": ["the groups retrieved from client authentication in Keycloak"]
+  }
+}
+```
+
+The authentication module will test if the authData is the same as the
+userinfo oauth call, comparing the attributes
```
Comment (TomWFox) [inline _includes/parse-server/third-party-auth.md]:
```suggestion
```

File: _includes/parse-server/third-party-auth.md [hunk 3]
```diff
@@ -189,6 +190,37 @@ Google oauth supports validation of id_token's and access_token's.
...
+  "keycloak": {
+    "access_token": "access token you got from keycloak JS client authentication",
+    "id": "the id retrieved from client authentication in Keycloak",
+    "roles": ["the roles retrieved from client authentication in Keycloak"],
+    "groups": ["the groups retrieved from client authentication in Keycloak"]
+  }
+}
+```
+
+The authentication module will test if the authData is the same as the
```
Comment (TomWFox) [inline _includes/parse-server/third-party-auth.md]:
```suggestion
The authentication module will test if the authData is the same as the userinfo oauth call, by comparing the attributes.
```

File: _includes/parse-server/third-party-auth.md [hunk 4]
```diff
@@ -189,6 +190,37 @@ Google oauth supports validation of id_token's and access_token's.
 }
 ```
 
+### Keycloak `authData`
+
+```js
+{
+  "keycloak": {
+    "access_token": "access token you got from keycloak JS client authentication",
```
Comment (TomWFox) [inline _includes/parse-server/third-party-auth.md]:
```suggestion
    "access_token": "access token from keycloak JS client authentication",
```

**Review by TomWFox — CHANGES_REQUESTED** [review]

**Comment by TomWFox** [comment]
Thanks for contributing!! Just a few small changes needed.

So you’re aware, this won’t be merged until a release of parse Server is made including your pr and I’ve made a final review.

File: _includes/parse-server/third-party-auth.md [hunk 5]
```diff
@@ -189,6 +190,36 @@ Google oauth supports validation of id_token's and access_token's.
...
+    "id": "the id retrieved from client authentication in Keycloak",
+    "roles": ["the roles retrieved from client authentication in Keycloak"],
+    "groups": ["the groups retrieved from client authentication in Keycloak"]
+  }
+}
+```
+
+The authentication module will test if the authData is the same as the userinfo oauth call, by comparing the attributes.
+
+Copy the JSON config file generated on Keycloak ([how-to link](https://www.keycloak.org/docs/latest/securing_apps/index.html#_javascript_adapter))
```
Comment (TomWFox) [inline _includes/parse-server/third-party-auth.md]:
```suggestion
Copy the JSON config file generated on Keycloak ([tutorial](https://www.keycloak.org/docs/latest/securing_apps/index.html#_javascript_adapter))
```

**Review by TomWFox — APPROVED** [review]
Minor suggestion but apart from that this looks good!

**Comment by rhuanbarreto** [comment]
https://github.com/parse-community/parse-server/pull/6376 was approved. You can also merge here.

**Pull request merged**